assets/src/components/ladder.tsx

```diff
@@ -40,113 +44,131 @@ const Ladder = ({
   selectedVehicleId,
 }: Props) => {
   const dispatch = useContext(DispatchContext)
-  const timepointSpacingY =
-    (HEIGHT - MARGIN_TOP_BOTTOM * 2) / (timepoints.length - 1)
+  const [height, setHeight] = useState(500)
+  const wrapperDivRef = useRef<HTMLDivElement>(null)
+  useLayoutEffect(() => {
+    if (wrapperDivRef.current !== null) {
+      const newHeight = wrapperDivRef.current.offsetHeight
+      if (newHeight !== height) {
+        setHeight(newHeight)
+      }
+    }
+  }, [wrapperDivRef.current, height])
 
   const orderedTimepoints: Timepoint[] =
     // Use slice to make a copy of the array before destructively reversing
     ladderDirection === LadderDirection.OneToZero
       ? timepoints.slice().reverse()
       : timepoints
 
+  const timepointSpacingY: number =
+    (height - MARGIN_TOP_BOTTOM * 2) / (timepoints.length - 1)
   const timepointStatusY = timepointStatusYFromTimepoints(
     timepoints,
     timepointSpacingY
   )
+
   const { ladderVehicles, widthOfLanes } = ladderVehiclesFromVehicles(
     vehicles,
     ladderDirection,
     timepointStatusY
   )
 
   const width = 120 + 2 * widthOfLanes
-  // (0, 0) is in the center of the first timepoint
-  const viewBox = [-width / 2, -MARGIN_TOP_BOTTOM, width, HEIGHT].join(" ")
+  const viewBox = [-width / 2, -MARGIN_TOP_BOTTOM, width, height].join(" ")
```

Comment:
Oops, I didn't mean to remove that comment `// (0, 0) is in the center of the first timepoint`. It's still relevant and probably worth keeping.